File: vipclaw-admin/src/main/java/com/vipclaw/admin/service/ModelService.java
package com.vipclaw.admin.service;

import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
import com.baomidou.mybatisplus.extension.service.IService;
import com.vipclaw.admin.dto.ModelCreateRequest;
import com.vipclaw.admin.dto.ModelResponse;
import com.vipclaw.admin.dto.ModelUpdateRequest;
import com.vipclaw.admin.entity.Model;

/**
 * 模型服务接口
 *
 * @author vipamp
 * @since 2026-03-13
 */
public interface ModelService extends IService<Model> {

    /**
     * 分页查询模型
     *
     * @param page       分页对象
     * @param name       名称
     * @param providerId 供应商ID
     * @param modelType  模型类型
     * @param status     状态
     * @param tags       标签筛选（支持多个，如：internet,reasoning,tool,mcp,vision）
     * @param minPrice   最低价格
     * @param maxPrice   最高价格
     * @return 分页结果
     */
    Page<ModelResponse> page(Page<Model> page, String name, Long providerId, String modelType, Integer status, String tags, Double minPrice, Double maxPrice);

    /**
     * 获取模型详情
     *
     * @param id ID
     * @return 模型响应
     */
    ModelResponse getDetail(Long id);

    /**
     * 创建模型
     *
     * @param request 创建请求
     * @return 模型响应
     */
    ModelResponse create(ModelCreateRequest request);

    /**
     * 更新模型
     *
     * @param id      ID
     * @param request 更新请求
     * @return 模型响应
     */
    ModelResponse update(Long id, ModelUpdateRequest request);

    /**
     * 切换模型状态
     *
     * @param id ID
     * @return 模型响应
     */
    ModelResponse toggle(Long id);
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/admin/service/ModelService.kt
package com.vipamp.vipclaw.admin.service

import com.vipamp.vipclaw.admin.dto.ModelCreateRequest
import com.vipamp.vipclaw.admin.dto.ModelResponse
import com.vipamp.vipclaw.admin.dto.ModelUpdateRequest
import com.vipamp.vipclaw.admin.dto.Page
import com.vipamp.vipclaw.admin.entity.Model

/**
 * Model service interface
 */
interface ModelService {

    /**
     * Query models with pagination
     *
     * @param name       Name
     * @param providerId Provider ID
     * @param modelType  Model type
     * @param status     Status
     * @param tags       Tags filter (supports multiple, e.g.: internet,reasoning,tool,mcp,vision)
     * @param minPrice   Minimum price
     * @param maxPrice   Maximum price
     * @param pageNum    Current page number
     * @param pageSize   Page size
     * @return Paginated result
     */
    fun page(
        name: String?,
        providerId: Long?,
        modelType: String?,
        status: Int?,
        tags: String?,
        minPrice: Double?,
        maxPrice: Double?,
        pageNum: Int,
        pageSize: Int,
    ): Page<Model>

    /**
     * Get model details
     *
     * @param id ID
     * @return Model entity
     */
    fun getModel(id: Long): Model?

    /**
     * Create model
     *
     * @param request Create request
     * @return Model response
     */
    fun createModel(request: ModelCreateRequest): Boolean

    /**
     * Update model
     *
     * @param id      ID
     * @param request Update request
     * @return Model response
     */
    fun updateModel(id: Long, request: ModelUpdateRequest): Boolean

    /**
     * Update model status
     *
     * @param id     ID
     * @param status Status (0:disabled, 1:enabled)
     * @return Update result
     */
    fun updateStatus(id: Long, status: Int): Boolean

    /**
     * Toggle model status
     *
     * @param id     ID
     * @param status Status (0:disabled, 1:enabled)
     * @return Update result
     */
    fun toggleModel(id: Long, status: Int): Boolean

    /**
     * Delete model (logical delete)
     *
     * @param id ID
     * @return Delete result
     */
    fun deleteModel(id: Long): Boolean

    /**
     * Convert model entity to model response
     *
     * @param model Model entity
     * @return Model response
     */
    fun convertToResponse(model: Model): ModelResponse
}
